iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -397,24 +396,14 @@ public override async Task SendTwinPatchAsync(TwinCollection reportedProperties,
 
                 Task<Twin> receivingTask = taskCompletionSource.Task;
 
-                if (await Task
-                    .WhenAny(receivingTask, Task.Delay(TimeSpan.FromSeconds(ResponseTimeoutInSeconds), cancellationToken))
-                    .ConfigureAwait(false) == receivingTask)
+                if (receivingTask.Exception?.InnerException != null)
```

Comment:
For this comment: https://github.com/Azure/azure-iot-sdk-csharp/pull/2653#discussion_r964141852